js/gateio.js

```diff
@@ -1312,41 +1314,82 @@ module.exports = class gateio extends Exchange {
     }
 
     async fetchFundingHistory (symbol = undefined, since = undefined, limit = undefined, params = {}) {
-        if (symbol === undefined) {
-            throw new ArgumentsRequired (this.id + ' fetchFundingHistory() requires a symbol argument');
-        }
         await this.loadMarkets ();
         // let defaultType = 'future';
-        const market = this.market (symbol);
-        const request = this.prepareRequest (market);
+        let market = undefined;
+        let request = {};
+        if (symbol !== undefined) {
+            market = this.market (symbol);
+            request = this.prepareRequest (market);
+        }
+        let type = undefined;
+        [ type, params ] = this.handleMarketTypeAndParams ('fetchFundingHistory', market, params);
+        if (market === undefined) {
+            const defaultSettle = (type === 'swap') ? 'usdt' : 'btc';
+            const settle = this.safeString (params, 'settle', defaultSettle);
+            request['settle'] = settle;
+            params = this.omit (params, 'settle');
+        }
         request['type'] = 'fund';  // 'dnw' 'pnl' 'fee' 'refr' 'fund' 'point_dnw' 'point_fee' 'point_refr'
         if (since !== undefined) {
-            request['from'] = since;
+            request['from'] = since / 1000;
         }
         if (limit !== undefined) {
             request['limit'] = limit;
         }
-        const method = this.getSupportedMapping (market['type'], {
+        const method = this.getSupportedMapping (type, {
             'swap': 'privateFuturesGetSettleAccountBook',
             'future': 'privateDeliveryGetSettleAccountBook',
         });
         const response = await this[method] (this.extend (request, params));
+        //
+        //    [
+        //        {
+        //            "time": 1646899200,
+        //            "change": "-0.027722",
+        //            "balance": "11.653120591841",
+        //            "text": "XRP_USDT",
+        //            "type": "fund"
+        //        },
+        //        ...
+        //    ]
+        //
+        return this.parseFundingHistories (response, symbol, since, limit);
```

Comment:
We must filter by `market['symbol']` instead of filtering by `symbol` otherwise it breaks filterBySymbolSinceLimit. Please, see these recent PRs for reference: https://github.com/ccxt/ccxt/pulls?q=is%3Apr+author%3Asc0Vu

Reply:
Adding this line 1323 would solve it:

```JavaScript
if (symbol !== undefined) {
    market = this.market (symbol);
    symbol = market['symbol']; // <<<<<<<<<<<<<<<<<<<<<<<<<
    request = this.prepareRequest (market);
}
```